src/ckanext-mappreview/ckanext/mappreview/plugin.py

```diff
@@ -28,7 +28,25 @@ def parse_metadata(pkg):
     mappreview = get_mappreview(pkg)
     if not mappreview:
         return {}
-    return json.loads(mappreview['value'])
+
+    metadata = json.loads(mappreview['value'])
+
+    # Extract dataset-level extras
+    center_lat_lon = None
+    for extra in pkg.get('extras', []):
+        if extra['key'] == 'center_lat_lon':
+            try:
+                center_lat_lon = json.loads(extra['value'])
+            except json.JSONDecodeError:
+                pass
+
+    # Inject into each vector layer (optional: only if one layer?)
+    for layer in metadata.get('layers', []):
+        if layer.get('type') == 'vector':
+            if center_lat_lon:
+                layer['center_lat_lon'] = center_lat_lon
```

Comment:
If I'm reading this right, this seems like it could raise an exception when `layers` is not in `metadata`.  This means `.get` returns `[]`, which we can't index into with the string `center_lat_lon`.  So maybe this should actually be returning a different type or something?